### ISSUE TITLE
Improve leaky abstraction: Gateway/Proxy

### ISSUE THREAD

**Issue by derrickburns** [solo-io/gloo#1910]
@ilackarms 

One of the challenges in using Gloo Gateway is the connection between the load balancer and the gateway proxy.  Let me explain.

External-Dns is a project that updates DNS Aliases based on Kubernetes Ingress resources and/or annotations on load balancer Kubernetes Services. This keeps DNS synced so that DNS aliases point to the proper load balancer that serves them.

In Gloo (Gateway), the information about which DNS names are served is in the Virtual Service.  The association between a Gateway and (one or more) gateway proxies is made in the Gateway resource.  However, the crucial information about which DNS names are served by the proxy is not made visible.  

In other words, one needs to create another mechanism to propagate that information to the Kubernetes Service. This mechanism is necessarily intrusive and global, and is therefore not composable.

Alternatively, one could have Gloo perform this function.  The most direct implementation has Gloo add annotations to the Kubernetes Service that are consistent with the format of External-Dns.  Another possibility is to add a status field to the Service, and to presume that another process will map from the status field to the annotation.  I am sure that there are other ways to address this as well.

**Comment by NelsonJeppesen** [comment]
+1

**Comment by ilackarms** [comment]
@NelsonJeppesen @derrickburns i believe this is a perfect use case for a Gloo Operator which manages the Service/Deployment for Gateway Proxy instances: #430
